sr_gui_controller_tuner/src/sr_gui_controller_tuner/sr_controller_tuner.py

```diff
@@ -91,63 +93,161 @@ def __init__(self, xml_path):
+                # if prefix matches the mapping, add this joint prefix
+                if prefix in value:
+                    hand_joint_prefixes.append(value)
+
+        if len(hand_joint_prefixes) > 1:
+            # the plugin cannot handle more than one hand
+            rospy.logwarn("More than one hand found with prefix :"+prefix+" !\n Not loading controllers")
+            return False
+
+        if len(hand_joint_prefixes) == 0:
+            rospy.logwarn("No hand found with prefix :"+prefix)
+            self.joint_prefix = ""
+        else:
+            self.joint_prefix = hand_joint_prefixes[0]
+
+        rospy.loginfo("using joint_prefix "+self.joint_prefix)
+        return True
 
     def get_ctrls(self):
         """
-        Retrieve currentlly running controllers
+        Retrieve currently running controllers
         return ["Motor Force", "Position"]
         """
-
         running_ctrls = []
 
-        try:
-            rospy.wait_for_service('controller_manager/list_controllers', self.CONTROLLER_MANAGER_DETECTION_TIMEOUT)
+        # find controller manager in selected namespace
 
-            controllers = rospy.ServiceProxy('controller_manager/list_controllers', ListControllers)
-            resp = None
-            try:
-                resp = controllers()
-            except rospy.ServiceException, e:
-                rospy.logerr( "Service did not process request: %s"%str(e) )
-
-            running_ctrls.append("Motor Force")
-            if resp != None:
-                for controller in resp.controller:
-                    if controller.state == "running":
-                        splitted = controller.name.split("_")
-                        ctrl_type_tmp = splitted[2]
+        ctrl_srv_name = self.prefix+'controller_manager/list_controllers'
+        try:
+            rospy.wait_for_service(ctrl_srv_name, self.CONTROLLER_MANAGER_DETECTION_TIMEOUT)
+
+        except rospy.ROSException:
+            # try at root namespace (only in case bimanual setup in a single loop and only if no GUI ns)
+            if self.namespace == "/":
+                ctrl_srv_name = 'controller_manager/list_controllers'
+                try:
+                    rospy.wait_for_service(ctrl_srv_name, self.CONTROLLER_MANAGER_DETECTION_TIMEOUT)
+                    self.single_loop = True
+                    rospy.loginfo("Detected single loop")
+                except rospy.ROSException, e:
+                    rospy.loginfo("Controller manager not running: %s"%str(e))
+                    rospy.loginfo("Running controller tuner in edit-only mode")
+                    self.edit_only_mode = True
+                    #In edit_only_mode all the controllers are available for editing
+                    for defined_ctrl_type in self.all_controller_types:
+                        running_ctrls.append(defined_ctrl_type)
+                    return running_ctrls
+            else:
+                self.edit_only_mode = True
```

Comment:
Seems to be small code duplication. If you'll have time you can extract small method from this code and use it in both places. I am personally using PyCharm IDE for Python http://wiki.ros.org/IDEs#PyCharm_.28community_edition.29 which has this refactoring pattern build in so to do this operation you'll need few clicks of the mouse or one short cut :) By the way it also has build in interactive PEP 8 checks.